src/python/grpcio/grpc/_channel.py

```diff
@@ -660,30 +665,27 @@ def __init__(self, channel):
     self.lock = threading.Lock()
     self.channel = channel
     self.completion_queue = cygrpc.CompletionQueue()
-    self.managed_calls = None
+    self.managed_calls = False
 
 
 def _run_channel_spin_thread(state):
   def channel_spin():
     while True:
-      event = state.completion_queue.poll()
-      completed_call = event.tag(event)
-      if completed_call is not None:
-        with state.lock:
-          state.managed_calls.remove(completed_call)
-          if not state.managed_calls:
-            state.managed_calls = None
-            return
+      try:
+        event = state.completion_queue.poll()
+        if event is not None:
+          event.tag(event)
+      except cygrpc.PollInterruptedException:
+        return
+        
 
   def stop_channel_spin(timeout):
-    with state.lock:
-      if state.managed_calls is not None:
-        for call in state.managed_calls:
-          call.cancel()
+    state.completion_queue.interrupt()
 
   channel_spin_thread = _common.CleanupThread(
       stop_channel_spin, target=channel_spin)
   channel_spin_thread.start()
+  state.managed_calls = True
```

Comment:
state.lock should already have been acquired by _create_channel_managed_call, but I agree its better to re-aquire here in case we use the function elsewhere.